arch/arc/Kconfig

```diff
@@ -18,10 +18,7 @@ config CPU_ARCEM
 
 config CPU_ARCHS
 	bool
-	# FIXME: ATOMIC_OPERATIONS_BUILTIN still has some problem in arcmwdt
-	# toolchain, so choosing ATOMIC_OPERATIONS_C instead.
-	select ATOMIC_OPERATIONS_C if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "arcmwdt"
-	select ATOMIC_OPERATIONS_BUILTIN if "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "arcmwdt"
+	select ATOMIC_OPERATIONS_BUILTIN
```

Comment:
We have builtin atomic fix starting from MWDT 2021.03, so we need to check that we have appropriate MWDT toolchain.

@IRISZZW could you please add MWDT version check and don't allow to use MWDT version < 2021.03. The check can be placed in `cmake/compiler/arcmwdt/target.cmake` where we do compiler discovery.

Reply:
PS: I'm totally OK if it is done as a follow-up pull request.

Reply:
I will do it in another pull request.